### ISSUE TITLE
Base profile doesn't work with docker

### ISSUE THREAD

**Issue by gomons** [conan-io/conan-package-tools#441]
* Conan Package Tools Version: **0.30.4**
* Operating System: **Linux**

### Steps to reproduce

1. Set CONAN_USE_DOCKER=1
1. Prepare conan config with profile foo
1. Pass config_url parameter with this config to ConanMultiPackager constructor
1. Pass foo profile name to ConanMultiPackager.run() method to make it base profile

**Actual result:**
cpt fails accessing None object

**Expected result:**
Works like a charm

### Additional info
As I can see, run_in_docker.py script is invoked in this case. There we can see line
`conan_api, client_cache, _ = Conan.factory()`, but factory method implementation looks like this:
````
class ConanAPIV1(object):
    @classmethod
    def factory(cls):
        return cls(), None, None
````
Below in the run_in_docker.py we can see condition:
````
if base_profile_text:
    base_profile_name = unscape_env(os.getenv("CPT_BASE_PROFILE_NAME"))
    tools.save(os.path.join(client_cache.profiles_path, base_profile_name),
               base_profile_text)
````
But client_cache here is None object which leads to the problem

**Comment by ghost** [comment]
We fixed this issue by applying the following patch locally:

```
--- a/cpt/run_in_docker.py
+++ b/cpt/run_in_docker.py
@@ -14,7 +14,11 @@ from cpt.uploader import Uploader
 
 def run():
     # Get all from environ
-    conan_api, client_cache, _ = Conan.factory()
+    conan_api, _, _ = Conan.factory()
+
+    conan_api.create_app()
+    client_cache = conan_api.app.cache
+
     printer = Printer()
 
     remotes_manager = RemotesManager(conan_api, printer)
```

**Comment by uilianries** [comment]
Fixed by #553

It should be available on CPT 0.35.1
